src/components/ArticleList.jsx

```diff
@@ -5,32 +5,45 @@ import Container from "react-bootstrap/Container";
 import Row from "react-bootstrap/Row";
 import Col from "react-bootstrap/Col";
 import './ArticleList.css'
-
-const endpoint = "https://nc-news-fswv.onrender.com/api/articles"
+import Loading from './Loading';
+import Error from './Error';
+import { fetchArticles } from '../../utils/api';
 
 export default function ArticleList() {
     
+    const [isLoading, setIsLoading] = useState(true)
+    const [isError, setIsError] = useState(false)
     const [articles, setArticles] = useState([]);
 
     useEffect(() => {
-        axios.get(endpoint)
+        setIsLoading(true)
+        fetchArticles()
         .then((response) => {
             setArticles(response.data.articles)
```

Comment:
You could handle getting articles off the response in the fetch articles function

Reply:
done